### PR TITLE
feat: Centralize host-to-provider & model-to-system Resolution Helpers

### PR DESCRIPTION
**Pull request by ialisaleh**
Refs #1739 

I added `get_provider_from_host` and `get_system_from_model` to the common package to centralize LLM provider and system inference helpers that were previously duplicated across multiple instrumentation packages.

This PR includes updated `OpenInferenceLLMProviderValues` and `OpenInferenceLLMSystemValues` enums with the full set of supported providers, data-driven prefix/suffix mapping dicts replacing ad-hoc `if/elif` chains, and a parametrized test suite covering all values and edge cases.

Once this is released, a follow-up PR will remove the duplicated helpers from each package in favor of importing from common.

### CODE REVIEW

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py [hunk 1]
```diff
@@ -57,6 +57,80 @@
...
+    "mistral": OpenInferenceLLMSystemValues.MISTRALAI.value,
+    "mixtral": OpenInferenceLLMSystemValues.MISTRALAI.value,
+    "pixtral": OpenInferenceLLMSystemValues.MISTRALAI.value,
+    "gemini": OpenInferenceLLMSystemValues.VERTEXAI.value,
+    "vertex": OpenInferenceLLMSystemValues.VERTEXAI.value,
+    "google": OpenInferenceLLMSystemValues.VERTEXAI.value,
+}
+
+
+def get_provider_from_host(host: str) -> Optional[str]:
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]:
```suggestion
def infer_llm_provider_from_host(host: str) -> Optional[OpenInferenceLLMProviderValues]:
```

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py [hunk 2]
```diff
@@ -74,6 +78,8 @@
     "suppress_tracing",
     "TraceConfig",
     "OITracer",
+    "HOST_SUFFIX_TO_PROVIDER",
+    "MODEL_PREFIX_TO_SYSTEM",
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/__init__.py]: These don't need to be exposed publicly

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py [hunk 3]
```diff
@@ -57,6 +57,80 @@
 if TYPE_CHECKING:
     from _typeshed import DataclassInstance
 
+# Maps hostname suffixes to their corresponding LLM provider value.
+HOST_SUFFIX_TO_PROVIDER: Dict[str, str] = {
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]:
```suggestion
_HOST_SUFFIX_TO_PROVIDER: Dict[str, str] = {
```

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py [hunk 4]
```diff
@@ -57,6 +57,80 @@
...
+    "api.fireworks.ai": OpenInferenceLLMProviderValues.FIREWORKS.value,
+    "api.moonshot.cn": OpenInferenceLLMProviderValues.MOONSHOT.value,
+    "api.cerebras.ai": OpenInferenceLLMProviderValues.CEREBRAS.value,
+    "api.perplexity.ai": OpenInferenceLLMProviderValues.PERPLEXITY.value,
+    "api.together.ai": OpenInferenceLLMProviderValues.TOGETHER.value,
+    "api.together.xyz": OpenInferenceLLMProviderValues.TOGETHER.value,
+}
+
+# Maps model name prefixes to their corresponding LLM system value.
+MODEL_PREFIX_TO_SYSTEM: Dict[str, str] = {
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]:
```suggestion
_MODEL_PREFIX_TO_SYSTEM: Dict[str, str] = {
```

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py [hunk 5]
```diff
@@ -57,6 +57,80 @@
...
+        return None
+
+    normalised = host.lower().strip()
+    for suffix, provider in HOST_SUFFIX_TO_PROVIDER.items():
+        if normalised.endswith(suffix):
+            return provider
+    return None
+
+
+def get_system_from_model(model_name: str) -> Optional[str]:
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]:
```suggestion
def infer_llm_system_from_model_name(model_name: str) -> Optional[OpenInferenceLLMSystemValues]:
```

File: python/openinference-instrumentation/tests/test_manual_instrumentation.py [hunk 6]
```diff
@@ -2958,3 +2967,141 @@ def test_pydantic_model_with_non_json_dict_keys(self) -> None:
...
+        ],
+    )
+    def test_unrecognised_model_returns_none(self, model_name: str) -> None:
+        assert get_system_from_model(model_name) is None
+
+    @pytest.mark.parametrize("model_name", [None, 42, [], {}])
+    def test_non_string_input_returns_none(self, model_name: object) -> None:
+        assert get_system_from_model(model_name) is None  # type: ignore[arg-type]
+
+    def test_every_system_has_at_least_one_prefix_entry(self) -> None:
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/tests/test_manual_instrumentation.py:3091]: Are these needed?

File: python/openinference-instrumentation/tests/test_manual_instrumentation.py [hunk 7]
```diff
@@ -2958,3 +2967,141 @@ def test_pydantic_model_with_non_json_dict_keys(self) -> None:
...
+        ],
+    )
+    def test_unrecognised_host_returns_none(self, host: str) -> None:
+        assert get_provider_from_host(host) is None
+
+    @pytest.mark.parametrize("host", [None, 42, [], {}])
+    def test_non_string_input_returns_none(self, host: object) -> None:
+        assert get_provider_from_host(host) is None  # type: ignore[arg-type]
+
+    def test_every_provider_has_at_least_one_host_entry(self) -> None:
```
Comment (axiomofjoy) [inline python/openinference-instrumentation/tests/test_manual_instrumentation.py:3024]: Are these needed?

**Review by axiomofjoy — APPROVED** [review]
Conditional approval pending requested changes

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py (hunk 1 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]: Addressed.

File: python/openinference-instrumentation/src/openinference/instrumentation/__init__.py (hunk 2 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/src/openinference/instrumentation/__init__.py]: Addressed.

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py (hunk 3 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]: Addressed.

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py (hunk 4 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]: Addressed.

File: python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py (hunk 5 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/src/openinference/instrumentation/_attributes.py]: Addressed.

File: python/openinference-instrumentation/tests/test_manual_instrumentation.py (hunk 6 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/tests/test_manual_instrumentation.py:3091]: I removed it, as type checking already prevents this at the call site.

File: python/openinference-instrumentation/tests/test_manual_instrumentation.py (hunk 7 above)
Comment (ialisaleh) [reply python/openinference-instrumentation/tests/test_manual_instrumentation.py:3024]: Yes, it catches the case where someone adds a new enum value but forgets to add a corresponding hostname entry.
